apps/librenmsdevices/librenms_devices.star

```diff
@@ -0,0 +1,213 @@
+                main_align = "space_evenly",
+                children = [
+                    render_header(),
+                    render.Marquee(
+                        width = 64,
+                        offset_start = 64,
+                        offset_end = 64,
+                        align = "center",
+                        child = render.Text(
+                            content = error_msg,
+                        ),
+                    ),
+                ],
+            ),
+        ),
+    )
+
+# Render a one pixel line
+def render_line(color):
+    return render.Box(
+        width = 64,
+        height = 1,
+        color = color,
+    )
+
+# Render a single box representing a device's status
+def render_box(color, size = BOX_SIZE):
+    colors = {"green": GREEN, "red": RED, "orange": ORANGE}
+    return render.Box(
+        height = size,
+        width = size,
+        color = colors[color],
+    )
+
+# Render rows of up/down data by querying LibreNMS server for device status
+def render_rows(devices):
+    boxes = list()
+    children = list()
+
+    for device in devices:
+        # Device is currently offline
+        if device["status"] == 0:
+            # See if the "Ignore Alerts" flag is set
+            if device["ignore"] == 1:
+                # Device is offline but set to ignore alerts, mark as green
+                boxes.append(render_box("green"))
+
+            else:
+                # Device is offline and not set to ignore alerts, mark as red
+                boxes.append(render_box("red"))
+
+        else:
+            # Device is up, and uptime information is present in device record
+            if device["uptime"] != None:
+                # Check if the device has been up for more than 24 hours
+                if int(device["uptime"]) < 86400:
+                    # Uptime is less than 24 hours, mark as orange
+                    boxes.append(render_box("orange"))
+
+                else:
+                    # Device uptime is more than 24 hours, mark as green
+                    boxes.append(render_box("green"))
+
+            else:
+                # Device has no uptime data but is up - mark as green
+                boxes.append(render_box("green"))
+
+        boxes.append(
+            render.Box(
+                height = BOX_SIZE,
+                width = 1,
+                color = "#000000",
+            ),
+        )
+
+    # Split the list into chunks
+    rows = [boxes[x:x + 23] for x in range(0, len(boxes), 23)]
+
+    for row in rows:
+        r = render.Row(
+            children = row,
+            expanded = False,
+            main_align = "center",
+        )
+        children.append(r)
+
+        padding_row = render.Row(
+            children = [
+                render_line(BLACK),
+            ],
+        )
+        children.append(padding_row)
+
+    return children
+
+def main(config):
+    children = list()
+
+    librenms_url = config.str("librenms_url") or ""
```

Comment:
It would be great if this case could produce a useful default graphic that demonstrates what the app will look like once it's fully installed. Gives users a sense of what to expect, before they start digging up urls and API keys and so on.